reference/shaders-msl/vulkan/frag/demote-to-helper.vk.nocompat.msl23.frag

```diff
@@ -5,8 +5,10 @@ using namespace metal;
 
 fragment void main0()
 {
+    bool spvIsDiscarded = false;
```

Comment:
I think a more direct way is that `bool spvIsDiscarded = simd_is_helper_thread();` and then that variable can be updated and passed down.